### PR TITLE
support for default printers, bug fix

### PR DESCRIPTION
**Pull request by TinyTrebuchet**
Added support of configurable user and system-wide default printers
- Default config directories are: "~/" and "/etc/cpdb"
- Custom config directors can be specified with: ./configure --with-user-configdir=<user_dir> --with-system-configdir=<system_dir>
- Four new commands added to demo/print_frontend.c:
  1. get-default-printer
  2. get-default-printer-for-backend <backend name>
  3. make-user-default-printer <printer id> <backend name>
  4. make-system-default-printer <printer id> <backend name>

Updated .gitignore file to include the correct files: "cpdb/..." instead of older "lib/...".

Fixed a memory bug where a possibly null pointer was freed without any check.

### CODE REVIEW

**Comment by TinyTrebuchet** [comment]
get-default-printer tries to find a default printer in the following order:
1. search in the system-wide config file
2. search in the user config file
3. fallback to CUPS default printer if CUPS backend exists
4. fallback to Save_As_PDF printer if FILE backend exists
5. fallback to default printer of any backend connected
6. return NULL if no printer is connected to the frontend

**Comment by tillkamppeter** [comment]
Could you change the commands `make-user-default-printer` and `make-system-default-printer` into `set-user-default-printer` and `set-system-default-printer`. Otherwise it seems to be all OK.

**Comment by tillkamppeter** [comment]
The priority order is also well chosen. I think there is no way to do it better.

**Comment by tillkamppeter** [comment]
Could you also rename any `make-default-...()` (or similar) function into `set-default-...()`?

**Comment by tillkamppeter** [comment]
And `cpdbMake...()` -> `cpdbSet...()`.

File: cpdb/cpdb-frontend.c [hunk 1]
```diff
@@ -492,6 +394,152 @@ cpdb_printer_obj_t *cpdbGetDefaultPrinter(cpdb_frontend_obj_t *f)
     return default_printer;
 }
 
+int cpdbMakeUserDefault(cpdb_printer_obj_t *p)
+{
```
Comment (tillkamppeter) [inline cpdb/cpdb-frontend.c:398]: `cpdbMakeUserDefault()` and `cpdbMakeSystemDefault()` create the same type of default printer file but only differ in the file mane under which they store the file. Here you could create one function for reading the file, adding the new entry and removing the duplicates, taking the file name and the printer object as parameters. Then you create two small wrapper functions `cpdbSetUserDefault()` and `cpdbSetSystemDefault()` calling your new function with the `CPDB_USER_DEFAULT_PRINTERS` and `CPDB_SYSTEM_DEFAULT_PRINTERS` resp. as first parameter and `p` as second parameter.

**Comment by TinyTrebuchet** [comment]
Okay, I willl update the function names and create the two wrapper functions.

**Comment by TinyTrebuchet** [comment]
Also, seems like I have accidentally inserted a tab at the start of every line in `cpdb/cpdb-frontend.h` in commit https://github.com/OpenPrinting/cpdb-libs/pull/10/commits/9d6bd57200e302c7d2c807398f55fff90f52dc49 . Will fix that too.

**Pull request merged**